### PR TITLE
Add job to run bin smoke test in VM based on legacy kernel (4.4)

### PR DESCRIPTION
**Pull request by xtremerui**
Implements https://github.com/concourse/ci/issues/344

We can verify if the bin-smoke-legacy job runs on a VM with older kernel, the behaviour mentioned in https://github.com/concourse/ci/issues/344 could be reproduced. And the test will hang forever. Ideally we'd prefer a way to either fail the test and/or output the garden-runc error. But at least for now if the job hangs one would know something is wrong and it'd better to check the logs of the smoke VM.

@jamieklassen @xtreme-sameer-vohra @xtreme-vikram-yadav

### CODE REVIEW

File: tasks/terraform-custom-image.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (aoldershaw) [inline tasks/terraform-custom-image.yml:1]: Should this task be removed? I'm guessing it was replaced by `tasks/packer-custom-image.yml`?

File: tasks/packer-custom-image.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+params:
+  GCP_PROJECT:
+  GCP_KEY:
+
+inputs:
+- name: ci
+- name: linux-kernel-44
+
+outputs:
+- name: outputs
```
Comment (aoldershaw) [inline tasks/packer-custom-image.yml:17]: Maybe use a less generic output name e.g. `packer-image`? And/or use `output_mapping` in the `task`?

File: tasks/scripts/packer-custom-image [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+cd ci/deployments/legacy-kernel
+
+echo "$GCP_KEY" > gcp.json
+export GOOGLE_APPLICATION_CREDENTIALS=gcp.json
+
+packer build \
+  -var region="us-west1" \
+  -var machine_type="n1-standard-1" \
+  -var zone="us-west1-b" \
```
Comment (aoldershaw) [inline tasks/scripts/packer-custom-image:20]: What are the implications of the region and machine type? Does it need to match exactly with the size/location of the VM it's used for?

If so, does the terraform need to be updated? https://github.com/concourse/ci/blob/3200d4f2ab4fca3a3501d77852606957d3b76f2f/deployments/legacy-kernel/main.tf#L3-L9

File: tasks/scripts/packer-custom-image [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -e -u
+
+image_name="u16k44"
+outputs=$PWD/image-name
```
Comment (aoldershaw) [inline tasks/scripts/packer-custom-image:6]:
```suggestion
```

File: tasks/scripts/packer-custom-image (hunk 3 above)
Comment (xtremerui) [reply tasks/scripts/packer-custom-image:20]: Interesting. The image that created in `us-west1` has a location `us`, from what I can see in gcp console. It seems out terraform smoke deployment that configured to `us-central1` can find it without problem. So I think as long as they are all located within US the regions/zones don't need to match.

Comment (aoldershaw) [reply tasks/scripts/packer-custom-image:19]: Cool, thanks. Also, is there a reason for going with `f1-micro`? Seems really small (1 vCPU, 0.6 GB memory)

Comment (xtremerui) [reply tasks/scripts/packer-custom-image:19]: this `main.tf` file is only used for testing if the packer created image could be used by terraform to create a VM. So it is for the script in https://github.com/concourse/ci/blob/15bc8429c18d9e894dae2b2c884e105d75120998/deployments/legacy-kernel/README.md.

**Pull request merged**